src/spec-transform-auto-layout.js

```diff
@@ -139,12 +146,20 @@ var getMaxTickLabelSize = function (domainValues, formatter, fnCalcTickLabelSize
         return size;
     }
 
-    var maxXTickText = _.max(domainValues, (x) => formatter(x).toString().length);
+    var maxXTickText;
+    let lastComputed = -Infinity;
+    domainValues.forEach((value) => {
+        let computed = formatter(value).toString().length;
+        if (computed > lastComputed || computed === -Infinity && maxXTickText === -Infinity) {
+            maxXTickText = value;
+            lastComputed = computed;
+        }
+    });
 
     // d3 sometimes produce fractional ticks on wide space
     // so we intentionally add fractional suffix
     // to foresee scale density issues
-    var suffix = _.isNumber(maxXTickText) ? '.00' : '';
+    var suffix = (Number.isFinite(maxXTickText) || Number.isNaN(maxXTickText)) ? '.00' : '';
```

Comment:
_.isNumber(NaN) returns true so I needed to add that condition to keep the same logic.

If it's not needed I can happily remove it.